### PR TITLE
[Task] DEP-412: RN 2.0.0

### PR DESCRIPTION
**Pull request by asklausen**
## Purpose
- Update RN to 2.0.0
- Support Android [3.1.1](https://github.com/getyoti/yoti-doc-scan-android/releases/tag/v3.1.1)
- Support iOS [4.0.1](https://github.com/getyoti/yoti-doc-scan-ios/releases/tag/v4.0.1)
- Allow integrators to choose which frameworks to include when using our RN wrapper (Android, iOS)
- General cleanup and improvements
- Updated READMEs

## External References (e.g. Jira / Zeplin / Confluence)
[DEP-412](https://lampkicking.atlassian.net/browse/DEP-412)

[DEP-412]: https://lampkicking.atlassian.net/browse/DEP-412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
+*.hmap
+*.ipa
+*.xcuserstate
+*.lock
+project.xcworkspace
+Pods/
+
+# Android/IJ
+#
+.idea
```
Comment (mircea-yoti) [inline .gitignore:30]: You could add a few more items for Android, such as `bin/`, `gen/`, etc. Let's take this offline and I can send you what we have set up internally for the native SDK.

File: LICENCE [hunk 2]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (mircea-yoti) [inline LICENCE:1]: Why was this removed?

**Comment by Zzhelyaz** [comment]
All major flows verified.

**Review by Zzhelyaz — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,97 +1,53 @@
...
 
-[![GitHub tag (latest SemVer)](https://img.shields.io/github/v/tag/getyoti/yoti-doc-scan-react-native?label=latest%20release)](https://github.com/getyoti/yoti-doc-scan-react-native/releases) [![Publish Release](https://github.com/getyoti/yoti-doc-scan-react-native/workflows/Publish%20Release/badge.svg)](https://github.com/getyoti/yoti-doc-scan-react-native/actions?query=workflow%3A%22Publish+Release%22)
-
-Yoti is an identity checking platform that allows organisations to verify who people are, online and in person. The Yoti Doc Scan SDK allows the user to take a photo of their identifying document which we verify instantly and prepare a response which your system can then retrieve. Further information can be found in the [documentation](https://developers.yoti.com/yoti/getting-started-docscan).
+A react native wrapper of Yoti IDV for [Android](https://github.com/getyoti/yoti-doc-scan-android) and [iOS](https://github.com/getyoti/yoti-doc-scan-ios). Yoti IDV allows a user of your app to take a photo of their document, as well as to scan or capture their face, we then verify this instantly and prepare a response, which your system can then retrieve on your hosted site.s
```
Comment (mircea-yoti) [inline README.md]: Small typo at the end

File: README.md [hunk 4]
```diff
@@ -101,162 +57,105 @@ If you're using **Proguard** or other obfuscation tool, add the following config
 -dontwarn javax.annotation.Nullable
 ```
 
-
-Depending on your Android project setup and version of React Native, you
```
Comment (mircea-yoti) [inline README.md:105]: Why were these troubleshooting steps removed?

File: example/package.json [hunk 5]
```diff
@@ -1,30 +1,21 @@
...
+    "name": "example",
+    "main": "index.js",
+    "private": true,
+    "scripts": {
+      "android": "react-native run-android",
+      "ios": "react-native run-ios",
+      "start": "react-native start"
+    },
+    "dependencies": {
+      "react": "^17.0.2",
```
Comment (mircea-yoti) [inline example/package.json:11]: I'm not a big fan of version ranges, since they might introduce changes we might not want/anticipate. 🤔

File: example/README.md [hunk 6]
```diff
@@ -1,27 +1,15 @@
...
-# Android (using React Native CLI)
-- Simply execute `yarn run android` in the `example` folder
+## Installation
+Install the wrapper's dependencies by running `yarn install` from the `example` directory, and then proceed to the relevant platform section below.
 
-# iOS (XCode)
-- Open the `ios/example.xcworkspace` file with XCode
-- Run a build as you regularly would, preferrably on a device with the Yoti app installed
+### Android
+To continue your installation with Android, you should run `yarn start` and `adb -s DEVICE_ID_HERE reverse tcp:8081 tcp:8081` (or `adb reverse tcp:8081 tcp:8081` if using a simulator) from the `example` directory. , and then open the `example/android` project in Android Studio. You can now sync Gradle and build the app.
```
Comment (mircea-yoti) [inline example/README.md]: A bit nitpicking here, but on Android it's "emulator" 😄

File: example/README.md [hunk 7]
```diff
@@ -1,27 +1,15 @@
...
+To run this example, a working working React Native environment is needed (see [reactnative.dev/docs/getting-started](https://reactnative.dev/docs/getting-started)).
 
-- In the example folder, run: `yarn start`
-- Open the `example/android` project in Android Studio
-- Sync Gradle
-- Run a build as you regularly would, preferrably on a device with the Yoti app installed
-- If you are running on a physical device, remember to execute `adb -s DEVICE_ID_HERE reverse tcp:8081 tcp:8081`. You can find the appropriate device id by running `adb devices`
-
-# Android (using React Native CLI)
-- Simply execute `yarn run android` in the `example` folder
```
Comment (mircea-yoti) [inline example/README.md:20]: This section is useful, it's way easier to simply run the app like this, I would not remove it.

File: example/android/app/build.gradle [hunk 8]
```diff
@@ -2,137 +2,29 @@ apply plugin: "com.android.application"
...
     }
-
     defaultConfig {
         applicationId "com.example"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 110
-        versionName "1.1.0"
+        versionCode 1
+        versionName "1.0.0"
```
Comment (mircea-yoti) [inline example/android/app/build.gradle:27]: I know that given this is an example app, it probably does not matter, but when downgrading `versionCode` and attempting to run (basically update) an app, you'd get an error saying you are attempting to install an older version and asking you to uninstall the previous newer one first.

Again, might not matter since this would never get published, but still thought I'd mention this.

File: example/android/app/build.gradle [hunk 9]
```diff
@@ -192,7 +76,12 @@ android {
 
 dependencies {
     implementation fileTree(dir: "libs", include: ["*.jar"])
-    implementation "com.facebook.react:react-native:0.68.5"  // From node_modules
+    implementation "com.facebook.react:react-native:0.68.5"
+    implementation "com.yoti.mobile.android.sdk:yoti-sdk-doc-scan:${rootProject.ext.yotiSdkVersion}"
```
Comment (mircea-yoti) [inline example/android/app/build.gradle:80]: Before we included all module deps, I'm wondering if we should change that now...

File: example/README.md (hunk 7 above)
Comment (asklausen) [reply example/README.md:20]: wasn't removed. just moved.

File: example/package.json (hunk 5 above)
Comment (asklausen) [reply example/package.json:11]: the range includes minors + patches. not majors, so it's fine

File: example/android/app/build.gradle (hunk 9 above)
Comment (asklausen) [reply example/android/app/build.gradle:80]: using id docs + liveness as default

File: LICENCE (hunk 2 above)
Comment (mircea-yoti) [reply LICENCE:1]: Nevermind, saw the updates to the README in the next commit 👍

File: .gitignore (hunk 1 above)
Comment (asklausen) [reply .gitignore:30]: not needed

File: LICENCE (hunk 2 above)
Comment (asklausen) [reply LICENCE:1]: because the licence is specified via url in the readme

File: example/android/app/build.gradle (hunk 9 above)
Comment (mircea-yoti) [reply example/android/app/build.gradle:80]: But liveness is commented below

File: README.md (hunk 4 above)
Comment (asklausen) [reply README.md:105]: because those steps are about general errors that can occur, and have nothing to do with our sdk specifically

File: example/android/app/build.gradle (hunk 9 above)
Comment (asklausen) [reply example/android/app/build.gradle:80]: its not

Comment (mircea-yoti) [reply example/android/app/build.gradle:80]: You're right, sorry, scrolled past it

**Review by mircea-yoti — APPROVED** [review]
